### PR TITLE
Fix blob versioning bug

### PR DESCRIPTION
**Pull request by mohsha-msft**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/testdata/perf/download_blob.go [hunk 1]
```diff
@@ -1,140 +0,0 @@
-// // Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (seankane-msft) [inline sdk/storage/azblob/testdata/perf/download_blob.go:1]: Can we leave the perf tests?

File: sdk/storage/azblob/zc_page_blob_client.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ func (pb PageBlobClient) WithSnapshot(snapshot string) PageBlobClient {
 // WithVersionID creates a new PageBlobURL object identical to the source but with the specified snapshot timestamp.
 // Pass "" to remove the version returning a URL to the base blob.
 func (pb PageBlobClient) WithVersionID(versionID string) PageBlobClient {
-	p := NewBlobURLParts(pb.URL())
+	p, _ := NewBlobURLParts(pb.URL())
```
Comment (seankane-msft) [inline sdk/storage/azblob/zc_page_blob_client.go:73]: We should check all of these errors

**Review by seankane-msft — APPROVED** [review]

File: sdk/storage/azblob/zc_page_blob_client.go (hunk 2 above)
Comment (mohsha-msft) [reply sdk/storage/azblob/zc_page_blob_client.go:73]: This for sure wouldn't be a problem. that's why I didn't check it. Problem occurs when customer provides us URL directly. In that case we do parsing error check inside the NewBlobURLParts() method.

**Comment by mohsha-msft** [comment]
FYI @zezha-msft

**Pull request merged**